### PR TITLE
Implement building multipart form content

### PR DESCRIPTION
**Pull request by crazytonyli**
### Description

`URLSession` doesn't have an API to build "multipart/form-data" content. This PR implements it ourselves. The new implementation is not used yet, but will be used when I work on replacing Alamofire with URLSession.

I don't think there are many optimizations we can do here, in terms of avoiding copying files. `URLRequest` only accepts `Data` or `InputStream` as its body. I had a though of subclassing `InputStream` to embed many `InputStream` instances internally and build the form-data content one the file when the `read` method is called. But its API is pretty strict: it can only be instantiated using a `Data` or `URL`, which seems to imply the file content behind the stream has to be ready for reading.

### Testing Details

See the unit tests. Let me know if you think there are more tests we can add.

---

- [x] Please check here if your pull request includes additional test coverage.
- [x] I have considered if this change warrants release notes and have added them to the appropriate section in the `CHANGELOG.md` if necessary.

### CODE REVIEW

File: WordPressKitTests/Utilities/MultipartFormTests.swift
Comment (mokagio) [inline WordPressKitTests/Utilities/MultipartFormTests.swift]: I suppose we can remove these `print` statements before merging, right?

```suggestion
```

File: WordPressKit/MultipartForm.swift
Comment (mokagio) [inline WordPressKit/MultipartForm.swift:28]: Have you considered removing `filename` and forcing callers to specify the whole path with `fileAtPath`?

**Review by mokagio — DISMISSED** [review]

File: WordPressKit/MultipartForm.swift
Comment (crazytonyli) [reply WordPressKit/MultipartForm.swift:28]: These two things are related but also should co-exist. The multipart form contains a `filename` field, whose value will be set by the `filename` argument here. We can use the file name of the  `fileAtPath` as the default value. But if some code passes a `filename` argument in, we should use it instead.

File: WordPressKitTests/Utilities/MultipartFormTests.swift
Comment (crazytonyli) [reply WordPressKitTests/Utilities/MultipartFormTests.swift]: Good call. They are gone in https://github.com/wordpress-mobile/WordPressKit-iOS/pull/654/commits/cab9ddf1046d83aa3d2045a89279f9ff6e352cc4.

**Review by mokagio — APPROVED** [review]

**Comment by mokagio** [comment]
Approved after @crazytonyli replied to my question about input parameters https://github.com/wordpress-mobile/WordPressKit-iOS/pull/654#discussion_r1446792796

**Pull request merged**
